crates/astria-merkle/src/lib.rs

```diff
@@ -0,0 +1,718 @@
+//! A RFC 6962 compliant merkle tree with a flat representation.
```

Comment:
maybe mention "except consistency proofs" since we don't need to implement those?